direct/tpm2/policy_test.go

```diff
@@ -281,6 +282,189 @@ func TestPolicyOrUpdate(t *testing.T) {
 	}
 }
 
+func TestPolicyPCRUpdate(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	// Use a trial session to calculate this policy
+	sess, cleanup2, err := PolicySession(thetpm, tpm.AlgSHA1, 16, Trial())
+	if err != nil {
+		t.Fatalf("setting up policy session: %v", err)
+	}
+	defer func() {
+		t.Helper()
+		if err := cleanup2(); err != nil {
+			t.Errorf("cleaning up policy session: %v", err)
+		}
+	}()
+
+	PCRs, err := CreatePCRSelection([]int{0, 1, 2, 3, 7})
+	if err != nil {
+		t.Fatalf("Failed to create PCRSelection")
+	}
+	selection := tpml.PCRSelection{
+		PCRSelections: []tpms.PCRSelection{
+			{
+				Hash:      tpm.AlgSHA1,
+				PCRSelect: PCRs,
+			},
+		},
+	}
+
+	pcrRead := PCRRead{
+		PCRSelectionIn: selection,
+	}
+
+	pcrReadRsp, err := pcrRead.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("Failed to read PCRSelection")
+	}
+
+	var expectedVal []byte
+	for _, digest := range pcrReadRsp.PCRValues.Digests {
+		expectedVal = append(expectedVal, digest.Buffer...)
```

Comment:
So this line seems fine an idiomatic to me.

`s1 = append(s1, s2...)` is the standard way to append the elements of s2 to s1. See https://go.dev/blog/slices-intro (search for "append one slice to another")

Reply:
Oops, I'm sorry. I made this comment on the wrong append line. Please see below, where we have

wrongCase := append(expectedVal, ...)

Since this will modify expectedVal if it needed to be reallocated, that could lead to problems.

Reply:
Okay that makes more sense. I will resolve this by making the wrongDigest a Hash of the expectedDigest.